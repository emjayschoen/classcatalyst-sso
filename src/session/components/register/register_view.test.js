import React from 'react';
import { shallow } from 'enzyme/build';
import RegisterView from './register_view';

const FOO_LOCATION = { state: { returnTo: '/foo/bar' } };
const NO_LOCATION = { state: {} };

describe('<RegisterView />', () => {
  let subject;

  describe('not logged in', () => {
    describe('without error', () => {
      beforeEach(() => {
        subject = shallow(<RegisterView signedIn={false} handleSubmit={jest.fn()} location={FOO_LOCATION} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('without location set', () => {
      beforeEach(() => {
        subject = shallow(<RegisterView signedIn={false} handleSubmit={jest.fn()} location={NO_LOCATION} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('with error', () => {
      beforeEach(() => {
        subject = shallow(<RegisterView signedIn={false} handleSubmit={jest.fn()} location={FOO_LOCATION} status="failed" />);
      });
      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });

  describe('when already logged in', () => {
    describe('with a returnTo location state set', () => {
      beforeEach(() => {
        subject = shallow(<RegisterView signedIn handleSubmit={jest.fn()} location={FOO_LOCATION} status={undefined} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('without a returnTo location state set', () => {
      beforeEach(() => {
        subject = shallow(<RegisterView signedIn handleSubmit={jest.fn()} location={NO_LOCATION} status={undefined} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});
