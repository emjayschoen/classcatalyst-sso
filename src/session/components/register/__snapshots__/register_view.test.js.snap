// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterView /> not logged in with error renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Sign Up
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Name"
      name="name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password"
      name="password"
      required={true}
      type="password"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password Confirmation"
      name="password_confirmation"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Register"
    />
    <span
      className="status"
    />
    <p>
      <Link
        to="/signin"
      >
        Already have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<RegisterView /> not logged in without error renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Sign Up
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Name"
      name="name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password"
      name="password"
      required={true}
      type="password"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password Confirmation"
      name="password_confirmation"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Register"
    />
    <span
      className="status"
    />
    <p>
      <Link
        to="/signin"
      >
        Already have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<RegisterView /> not logged in without location set renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Sign Up
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Name"
      name="name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password"
      name="password"
      required={true}
      type="password"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password Confirmation"
      name="password_confirmation"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Register"
    />
    <span
      className="status"
    />
    <p>
      <Link
        to="/signin"
      >
        Already have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<RegisterView /> when already logged in with a returnTo location state set renders correctly 1`] = `
<Redirect
  to="/foo/bar"
/>
`;

exports[`<RegisterView /> when already logged in without a returnTo location state set renders correctly 1`] = `
<Redirect
  to="/dashboard"
/>
`;
